scripts/auth-common.sh

```diff
@@ -16,6 +14,24 @@ fi
 
 IRONIC_HTPASSWD_FILE=/etc/ironic/htpasswd
 INSPECTOR_HTPASSWD_FILE=/etc/ironic-inspector/htpasswd
+yaml() {
+        python3 -c "import yaml;
+try:
+    print(yaml.safe_load(open('$1'))$2)
+except:
+    print('')"
+}
+
+IRONIC_HTPASSWD=$(yaml ${IRONIC_HTPASSWD_FILE} "['data']['htpasswd']")
```

Comment:
You're trying to read the file you will be writing (also, it's not YAML).

If you mount a secret to, say, `/auth/ironic`, you'll get a directory with files corresponding to every secret key. In you case, you'll get `/auth/ironic/username` and `/auth/ironic/password`. You need to take these and make htpasswd out of them (check the man pages of the htpasswd tool).

Reply:
Reference https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#create-a-pod-that-has-access-to-the-secret-data-through-a-volume